api/lql_test.go

```diff
@@ -0,0 +1,632 @@
+}`,
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				StartTimeRange: "1970-01-01T00:11:18Z",
+				EndTimeRange:   "1973-07-11T04:32:21Z",
+				QueryText:      "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+				QueryBlob: `{
+	"start_time_range": "678910",
+	"end_time_range": "111213141516",
+	"query_text": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+		},
+		LQLQueryTest{
+			Name: "lql-blob",
+			Input: &api.LQLQuery{
+				QueryBlob: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				QueryText: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+				QueryBlob: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+			},
+		},
+		LQLQueryTest{
+			Name: "overwrite-blob",
+			Input: &api.LQLQuery{
+				StartTimeRange: "0",
+				EndTimeRange:   "1",
+				QueryText:      "should not overwrite",
+				QueryBlob: `{
+	"START_TIME_RANGE": "678910",
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				StartTimeRange: "1970-01-01T00:00:00Z",
+				EndTimeRange:   "1970-01-01T00:00:00Z",
+				QueryText:      "should not overwrite",
+				QueryBlob: `{
+	"START_TIME_RANGE": "678910",
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+		},
+	}
+)
+
+type LQLTranslateTimeTest struct {
+	Name       string
+	Input      string
+	ReturnTime string
+	ReturnErr  interface{}
+}
+
+type LQLValidateRangeTest struct {
+	Name       string
+	Input      api.LQLQuery
+	AllowEmpty bool
+	Return     interface{}
+}
+
+type LQLValidateTest struct {
+	Name   string
+	Input  *api.LQLQuery
+	Return interface{}
+}
+
+type LQLQueryTest struct {
+	Name     string
+	Input    *api.LQLQuery
+	Return   interface{}
+	Expected *api.LQLQuery
+}
+
+func TestTranslateTime(t *testing.T) {
+	for _, lqlTranslateTimeTest := range lqlTranslateTimeTests {
+		t.Run(lqlTranslateTimeTest.Name, func(t *testing.T) {
+			outTime, err := api.LQLQuery{}.TranslateTime(lqlTranslateTimeTest.Input)
+			if err == nil {
+				assert.Equal(t, lqlTranslateTimeTest.ReturnTime, outTime)
+				assert.Equal(t, lqlTranslateTimeTest.ReturnErr, err)
+			} else {
+				assert.Equal(t, lqlTranslateTimeTest.ReturnErr, err.Error())
+			}
+		})
+	}
+}
+
+func TestValidateRange(t *testing.T) {
+	for _, lqlValidateRangeTest := range lqlValidateRangeTests {
+		t.Run(lqlValidateRangeTest.Name, func(t *testing.T) {
+			err := lqlValidateRangeTest.Input.Translate()
+			assert.Nil(t, err)
+			err = lqlValidateRangeTest.Input.ValidateRange(lqlValidateRangeTest.AllowEmpty)
+			if err == nil {
+				assert.Equal(t, lqlValidateRangeTest.Return, err)
+			} else {
+				assert.Equal(t, lqlValidateRangeTest.Return, err.Error())
+			}
+		})
+	}
+}
+
+func TestValidate(t *testing.T) {
```

Comment:
This test function name is too general, there could be a number of `Validate()` functions that we can test, this one, in particular, is testing the validation inside the LQL query, let's be specific to avoid collisions in the future.
```suggestion
func TestLQLQueryValidate(t *testing.T) {
```
We should also change these functions to reference the `LQL` service,
otherwise we won't know what are we testing:

- func TestTranslateTime(t *testing.T) {
- func TestValidateRange(t *testing.T) {
- func TestCreateMethod(t *testing.T) {
- func TestCreateBadInput(t *testing.T) {
- func TestCreateOK(t *testing.T) {
- func TestCreateError(t *testing.T) {
- func TestGetQueriesMethod(t *testing.T) {
- func TestGetQueryByIDOK(t *testing.T) {
- func TestGetQueryByIDNotFound(t *testing.T) {
- func TestRunQueryMethod(t *testing.T) {
- func TestRunQueryBadInput(t *testing.T) {
- func TestRunQueryOK(t *testing.T) {
- func TestRunQueryError(t *testing.T) {

A suggestion would be to start the test function to `TestLQL...`

Reply:
Fixed.